src/components/HistoricWorthChart.tsx

```diff
@@ -0,0 +1,156 @@
+import { VictoryArea } from 'victory-native'
+
+import { useAppSelector } from '~/hooks/redux'
+import { selectHaveHistoricBalancesLoaded, selectIsStateUninitialized } from '~/store/addresses/addressesSelectors'
+import { selectAllAddresses } from '~/store/addressesSlice'
+import { useGetHistoricalPriceQuery } from '~/store/assets/priceApiSlice'
+import { ChartLength, DataPoint, LatestAmountPerAddress } from '~/types/charts'
+import { Currency } from '~/types/settings'
+
+interface HistoricWorthChart {
+  length?: ChartLength
+  latestWorth: number
+  currency: Currency
+  onWorthInBeginningOfChartChange: (worthInBeginningOfChart?: DataPoint['worth']) => void
+}
+
+const now = dayjs()
+const startingDates: Record<ChartLength, Dayjs> = {
+  '1d': now.subtract(1, 'day'),
+  '1w': now.subtract(1, 'week'),
+  '1m': now.subtract(1, 'month'),
+  '1y': now.subtract(1, 'year')
+}
+
+const HistoricWorthChart = ({
+  length = '1m',
+  latestWorth,
+  currency,
+  onWorthInBeginningOfChartChange
+}: HistoricWorthChart) => {
+  const theme = useTheme()
+  const { data: alphPriceHistory } = useGetHistoricalPriceQuery({ currency, days: 365 })
+  const addresses = useAppSelector(selectAllAddresses)
+  const haveHistoricBalancesLoaded = useAppSelector(selectHaveHistoricBalancesLoaded)
+  const stateUninitialized = useAppSelector(selectIsStateUninitialized)
+
+  const [chartData, setChartData] = useState<DataPoint[]>([])
+
+  const startingDate = startingDates[length].format('YYYY-MM-DD')
+  const isDataAvailable = addresses.length !== 0 && haveHistoricBalancesLoaded && !!alphPriceHistory
+  const filteredChartData = getFilteredChartData(chartData, startingDate)
+  const firstItem = filteredChartData.at(0)
+
+  useEffect(() => {
+    onWorthInBeginningOfChartChange(firstItem?.worth)
+  }, [firstItem?.worth, onWorthInBeginningOfChartChange])
+
+  useEffect(() => {
+    if (!isDataAvailable) {
+      setChartData([])
+      return
+    }
+
+    const computeChartDataPoints = (): DataPoint[] => {
+      const addressesLatestAmount: LatestAmountPerAddress = {}
+
+      return alphPriceHistory.map(({ date, price }) => {
+        let totalAmountPerDate = BigInt(0)
+
+        addresses.forEach(({ hash, balanceHistory }) => {
+          const amountOnDate = balanceHistory.entities[date]?.balance
+
+          if (amountOnDate !== undefined) {
+            const amount = BigInt(amountOnDate)
+            totalAmountPerDate += amount
+            addressesLatestAmount[hash] = amount
+          } else {
+            totalAmountPerDate += addressesLatestAmount[hash] ?? BigInt(0)
+          }
+        })
+
+        return {
+          date,
+          worth: price * parseFloat(toHumanReadableAmount(totalAmountPerDate))
+        }
+      })
+    }
+
+    const trimInitialZeroDataPoints = (data: DataPoint[]) => data.slice(data.findIndex((point) => point.worth !== 0))
+
+    let dataPoints = computeChartDataPoints()
+    dataPoints = trimInitialZeroDataPoints(dataPoints)
```

Comment:
Nit: 

```ts
const dataPoints = trimInitialZeroDataPoints(computeChartDataPoints())
```

Also, `computeChartDataPoints` is only used once and doesn't need to be a function?

Reply:
Both `computeChartDataPoints` and `trimInitialZeroDataPoints` are called only once. The reason I made them functions is to use the function name as a hint to the developer of what is going on there so that I don't have to add a comment. I think it improves code readability.

Reply:
Yes, I agree that declaring `trimInitialZeroDataPoints ` makes it more readable. IMO `chartDataPoints` could be a variable, but I'm 💯%  fine if we keep a function for this.

Reply:
How about dis? :) https://github.com/alephium/mobile-wallet/pull/141/commits/294c42d8771774cd9e67cfbf09a4b658e0c2f597

Reply:
Yes! Just be careful, you have a styled component hidden in between to utils functions!

Reply:
oopsi!